ts/Dashboards/Plugins/NavigatorComponent.ts

```diff
@@ -0,0 +1,690 @@
+            height = this.contentElement.clientHeight,
+            width = this.contentElement.clientWidth,
+            chartUpdates: HighchartsOptions = {};
+
+        if (
+            chart.chartHeight !== height ||
+            chart.chartWidth !== width
+        ) {
+            chartUpdates.chart = {
+                height,
+                width
+            };
+        }
+
+        if (chart.navigator) {
+            const navigator = chart.navigator,
+                navigatorHeight =
+                    (navigator.top - chart.plotTop + navigator.height);
+
+            if (navigator.height !== navigatorHeight) {
+                chartUpdates.navigator = {
+                    handles: {
+                        height: Math.round(height / 4)
+                    },
+                    height: navigatorHeight
+                };
+            }
+
+            chart.update(chartUpdates, false);
+
+            if (navigator.series && navigator.series[0]) {
+                navigator.series[0].update({
+                    type: chart.series[0].type
+                }, false);
+            }
+        } else {
+            chart.update(chartUpdates, false);
+        }
+    }
+
+
+    /**
+     * Returns the first column of columnAssignments to use for navigator data.
+     * @private
+     *
+     * @return
+     * Navigator column assignment.
+     */
+    public getColumnAssignment(): [string, string] {
+        const columnAssignments = (this.options.columnAssignments || {});
+
+        let columnsAssignment: (string|null);
+
+        for (const column of Object.keys(columnAssignments)) {
+            columnsAssignment = columnAssignments[column];
+
+            if (columnsAssignment !== null) {
+                return [column, columnsAssignment];
+            }
+        }
+
+        if (this.connector) {
+            const columns = this.connector.table.getColumnNames();
+
+            if (columns.length) {
+                return [columns[0], 'y'];
+            }
+        }
+
+        return ['', 'y'];
+    }
+
+
+    /**
+     * Gets the component's options.
+     * @internal
+     */
+    public getOptions(): Partial<NavigatorComponentOptions> {
+        return {
+            ...diffObjects(this.options, NavigatorComponentDefaults),
+            type: 'Navigator'
+        };
+    }
+
+
+    /** @private */
+    public load(): this {
+
+        super.load();
+
+        this.contentElement.appendChild(this.chartContainer);
+        this.parentElement.appendChild(this.element);
+        this.hasLoaded = true;
+
+        return this;
+    }
+
+
+    /** @private */
+    private redrawChart(): void {
+        const timeouts = this.resizeTimeouts;
+
+        for (let i = 0, iEnd = timeouts.length; i < iEnd; ++i) {
+            clearTimeout(timeouts[i]);
+        }
+
+        timeouts.length = 0;
+
+        timeouts.push(setTimeout((): void => {
+            this.adjustNavigator();
+            this.chart.redraw();
+        }, 50));
```

Comment:
Why 50ms? It doesn't look good on the initial render

Reply:
Navigator animation can not be turned off. I will try an increase.